gordon/interfaces.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from zope.interface import Attribute
+from zope.interface import Interface
+
+
+class IEventMessage(Interface):
+    """A discrete unit of work for Gordon to process.
+
+    Gordon expects plugins to return or accept objects that implement this
+    interface in order to route them to other plugins, and handle retries
+    or cleanup in case of errors.
+    """
+
+    def __init__(msg_id, data, phase=None, history_log=None):
```

Comment:
Will there ever be a time where a new event message will be initialized with some content in the history log? I'm wondering if there should be some `setter`/getter functionality instead that the `append_to_history` makes use of (I don't know, just typing half-baked/uncaffeinated ideas).

Reply:
There might be a case when it's init'ed with eg. `[f'Created msg at {datetime.now()}']`, but I think that the usual case will involve init'ing it with an empty list. I'd prefer this to setter/getter methods to keep the interface leaner, unless you feel strongly about this?

Reply:
Ah, hmm. Well first, "...the usual case will involve init'ing it with an empty list" is not true when you have `history_log=None`. Granted, it shouldn't be `history_log=[]` in the args. But it's not necessarily obvious to a plugin that an event message should be initialed with an empty list. 

Yeah I don't know the best approach, so let's leave it at this and change later if we want.

Reply:
Good point @econchick, I switched the `history_log` arg into being a required param, which should make it stand out more (and the fact that it should be a mutable sequence).